backend/typescript/models/scheduling.model.ts

```diff
@@ -0,0 +1,35 @@
+import { Column, DataType, Model, Table } from "sequelize-typescript";
+import { Status } from "../types";
+
+@Table({ tableName: "scheduling" })
+export default class Scheduling extends Model {
+  @Column({ type: DataType.INTEGER, primaryKey: true })
+  id!: number;
+
+  @Column({ type: DataType.INTEGER })
+  donor_id!: string;
+
+  @Column({ type: DataType.TEXT })
+  description!: string | null;
+
+  @Column({ type: DataType.INTEGER })
+  quantity!: number | null;
+
+  @Column({ type: DataType.DATE })
+  start_time!: Date;
+
+  @Column({ type: DataType.DATE })
+  end_time!: Date;
+
+  @Column({ type: DataType.ENUM("Rejected", "Approved", "Pending") })
+  status!: Status;
+
+  @Column({ type: DataType.INTEGER })
+  volunteers_needed!: number;
+
+  @Column({ type: DataType.ARRAY(DataType.INTEGER) })
+  volunteer_ids!: number[] | null;
```

Comment:
Once we have a volunteer table, volunteer_id would be a foreign key reference to the volunteer table. I am not sure if we can have a list of foreign key references. We might have to create another table `schedule_volunteer_mapping` which maps volunteer id to scheduling id in the future. For now, we can probably keep it like this since we dont have a volunteer table, but if you can leave a comment about this, that would be nice.